istioctl/cmd/vm-bootstrap.go

```diff
@@ -0,0 +1,630 @@
+		if err != nil {
+			return err
+		}
+		if sshPassword == "" {
+			return fmt.Errorf("a password, or ssh key location is required for vm-bootstrap")
+		}
+		sshAuthMethod = ssh.Password(sshPassword)
+	} else {
+		// Attempt to parse the key.
+		rawKey, err := ioutil.ReadFile(sshKeyLocation)
+		if err != nil {
+			return err
+		}
+		key, err := ssh.ParsePrivateKey(rawKey)
+		if err != nil {
+			if err, ok := err.(*ssh.PassphraseMissingError); ok {
+				term := terminal.NewTerminal(os.Stdin, "")
+				sshKeyPassword, err := term.ReadPassword("Please enter the password for the ssh key: ")
+				if err != nil {
+					return err
+				}
+				decryptedKey, err := ssh.ParsePrivateKeyWithPassphrase(rawKey, []byte(sshKeyPassword))
+				if err != nil {
+					return err
+				}
+				sshAuthMethod = ssh.PublicKeys(decryptedKey)
+			} else {
+				return err
+			}
+		} else {
+			sshAuthMethod = ssh.PublicKeys(key)
+		}
+	}
+
+	return nil
+}
+
+func vmBootstrapCommand() *cobra.Command {
+	vmBSCommand := &cobra.Command{
+		Use:   "vm-bootstrap",
+		Short: "bootstraps a workloadentry representing a vm into the mesh",
+		Args: func(cmd *cobra.Command, args []string) error {
+			if (len(args) == 1) == all {
+				cmd.Println(cmd.UsageString())
+				return fmt.Errorf("vm-bootstrap requires a workload entry, or the all flag")
+			}
+			if all && workloadNamespace == "" {
+				return fmt.Errorf("vm-bootstrap needs a namespace if fetching all workspaces")
+			}
+			if sshUser == "" {
+				user, err := user.Current()
+				if err != nil {
+					return err
+				}
+				sshUser = user.Username
+			}
+			if dumpDir == "" {
+				err := deriveSSHMethod()
+				if err != nil {
+					return err
+				}
+			}
+
+			return nil
+		},
+		RunE: func(c *cobra.Command, args []string) error {
+			var configClient model.ConfigStore
+			var err error
+
+			if configClient, err = clientFactory(); err != nil {
+				return err
+			}
+
+			var entries []model.Config
+			var chosenNS string
+			if all {
+				entries, chosenNS, err = fetchAllWorkloadEntries(configClient)
+			} else {
+				entries, chosenNS, err = fetchSingleWorkloadEntry(args[0], configClient)
+			}
+			if err != nil {
+				return err
+			}
+
+			kubeClient, err := interfaceFactory(kubeconfig)
+			if err != nil {
+				return err
+			}
+
+			certs, err := getCertificate(kubeClient)
+			if err != nil {
+				return err
+			}
+
+			addresses, err := getCertificatesForEachAddress(entries, chosenNS, certs)
+			if err != nil {
+				return err
+			}
+
+			if dumpDir != "" {
+				err = dumpCertificates(dumpDir, addresses)
+				if err != nil {
+					return err
+				}
+			} else {
+				err = copyCertificates(kubeClient, addresses)
+				if err != nil {
+					return err
+				}
+			}
+
+			return nil
+		},
+	}
+
+	vmBSCommand.PersistentFlags().BoolVarP(&all, "all", "a", false,
+		"attempt to bootstrap all workload entries")
+	vmBSCommand.PersistentFlags().StringVarP(&dumpDir, "dump-directory", "d", "",
```

Comment:
Maybe `--target-dir` or `--local-dir` instead of `--dump-directory`?  Also, perhaps add all of the options implemented to https://docs.google.com/document/d/17zQ5uN3DEagl6y65B1qZXWBHXAmjqiRWXicim92FAMU/edit# or label them experimental in the source code?

Reply:
Will change to: `--local-certs-dir`.

As for the options I didn't realize you could mark options experimental, and most of these definitely are. Even more so if we stop using certs which has been suggested a couple times.